File: java/Cpu.java
class Cpu extends Komponen implements PerangkatTambahan {
  private int jumlahCore;
  private double kecepatanGHz;

  public Cpu(int jumlahCore, double kecepatanGHz, String merk, String nama) {
      super(merk, nama);
      this.jumlahCore = jumlahCore;
      this.kecepatanGHz = kecepatanGHz;
  }

  @Override
  public void info() {
      System.out.println("CPU: " + merk + " " + nama + " (" + jumlahCore + " Core, " + kecepatanGHz + " GHz)");
  }

  @Override
  public void infoTambahan() {
      System.out.println("Tambahan CPU: Overclocking support.");
  }
}

File: python/cpu.py
from komponen import Komponen
from perangkat_tambahan import PerangkatTambahan

class Cpu(Komponen, PerangkatTambahan):
    def __init__(self, jumlah_core: int, kecepatan_ghz: float, merk: str, nama: str):
        super().__init__(merk, nama)
        self.jumlah_core = jumlah_core
        self.kecepatan_ghz = kecepatan_ghz

    def info(self):
        print(f"CPU: {self.merk} {self.nama} ({self.jumlah_core} Core, {self.kecepatan_ghz} GHz)")

    def info_tambahan(self):
        print("Tambahan CPU: Overclocking support.")
